TMUIPlugin.start() throws SWTError when not called from UI thread

In our LSP tests based on JDT LS, we sometimes see this logged error:

```
!ENTRY org.eclipse.osgi 4 0 2026-06-04 03:41:49.289
!MESSAGE An error occurred while automatically activating bundle org.eclipse.tm4e.ui (590).
!STACK 0
org.osgi.framework.BundleException: Exception in org.eclipse.tm4e.ui.TMUIPlugin.start() of bundle org.eclipse.tm4e.ui.
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:859)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:780)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:1082)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:394)
	at org.eclipse.osgi.container.Module.doStart(Module.java:643)
	at org.eclipse.osgi.container.Module.start(Module.java:500)
	at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:528)
	at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:122)
	at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:620)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:353)
	at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:418)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:527)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:439)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:195)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:490)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.loadClass(EquinoxBundle.java:663)
	at org.eclipse.core.internal.registry.osgi.RegistryStrategyOSGI.createExecutableExtension(RegistryStrategyOSGI.java:226)
	at org.eclipse.core.internal.registry.ExtensionRegistry.createExecutableExtension(ExtensionRegistry.java:1034)
	at org.eclipse.core.internal.registry.ConfigurationElement.createExecutableExtension(ConfigurationElement.java:286)
	at org.eclipse.core.internal.registry.ConfigurationElementHandle.createExecutableExtension(ConfigurationElementHandle.java:65)
	at org.eclipse.core.internal.filebuffers.ExtensionsRegistry.getExtension(ExtensionsRegistry.java:228)
	at org.eclipse.core.internal.filebuffers.ExtensionsRegistry.doGetDocumentSetupParticipants(ExtensionsRegistry.java:360)
	at org.eclipse.core.internal.filebuffers.ExtensionsRegistry.getDocumentSetupParticipants(ExtensionsRegistry.java:384)
	at org.eclipse.core.internal.filebuffers.ResourceExtensionRegistry.getDocumentSetupParticipants(ResourceExtensionRegistry.java:131)
	at org.eclipse.core.internal.filebuffers.ResourceTextFileBufferManager.createEmptyDocument(ResourceTextFileBufferManager.java:149)
	at org.eclipse.core.internal.filebuffers.ResourceTextFileBuffer.initializeFileBufferContent(ResourceTextFileBuffer.java:246)
	at org.eclipse.core.internal.filebuffers.ResourceFileBuffer.create(ResourceFileBuffer.java:248)
	at org.eclipse.core.internal.filebuffers.TextFileBufferManager.connect(TextFileBufferManager.java:118)
	at org.eclipse.jdt.ls.core.internal.handlers.JsonRpcHelpers.toDocument(JsonRpcHelpers.java:203)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceDiagnosticsHandler.visit(WorkspaceDiagnosticsHandler.java:201)
	at org.eclipse.core.internal.events.ResourceDelta.accept(ResourceDelta.java:78)
	at org.eclipse.core.internal.events.ResourceDelta.accept(ResourceDelta.java:92)
	at org.eclipse.core.internal.events.ResourceDelta.accept(ResourceDelta.java:92)
	at org.eclipse.core.internal.events.ResourceDelta.accept(ResourceDelta.java:61)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceDiagnosticsHandler.resourceChanged(WorkspaceDiagnosticsHandler.java:114)
	at org.eclipse.core.internal.events.NotificationManager$1.run(NotificationManager.java:342)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.core.internal.events.NotificationManager.notify(NotificationManager.java:331)
	at org.eclipse.core.internal.events.NotificationManager.broadcastChanges(NotificationManager.java:185)
	at org.eclipse.core.internal.resources.Workspace.broadcastPostChange(Workspace.java:473)
	at org.eclipse.core.internal.resources.Workspace.endOperation(Workspace.java:1619)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2522)
	at org.eclipse.core.internal.events.NotificationManager$NotifyJob.run(NotificationManager.java:68)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: org.eclipse.swt.SWTError: Invalid thread access
	at org.eclipse.ui.plugin.AbstractUIPlugin.createImageRegistry(AbstractUIPlugin.java:173)
	at org.eclipse.ui.plugin.AbstractUIPlugin.getImageRegistry(AbstractUIPlugin.java:219)
	at org.eclipse.tm4e.ui.TMUIPlugin.start(TMUIPlugin.java:174)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:838)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:1)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:251)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:830)
	... 43 more
Root exception:
org.eclipse.swt.SWTError: Invalid thread access
	at org.eclipse.ui.plugin.AbstractUIPlugin.createImageRegistry(AbstractUIPlugin.java:173)
	at org.eclipse.ui.plugin.AbstractUIPlugin.getImageRegistry(AbstractUIPlugin.java:219)
	at org.eclipse.tm4e.ui.TMUIPlugin.start(TMUIPlugin.java:174)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:838)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:1)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:251)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:830)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:780)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:1082)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:394)
	at org.eclipse.osgi.container.Module.doStart(Module.java:643)
	at org.eclipse.osgi.container.Module.start(Module.java:500)
	at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:528)
	at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:122)
	at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:620)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:353)
	at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:418)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:527)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:439)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:195)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:490)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.loadClass(EquinoxBundle.java:663)
	at org.eclipse.core.internal.registry.osgi.RegistryStrategyOSGI.createExecutableExtension(RegistryStrategyOSGI.java:226)
	at org.eclipse.core.internal.registry.ExtensionRegistry.createExecutableExtension(ExtensionRegistry.java:1034)
	at org.eclipse.core.internal.registry.ConfigurationElement.createExecutableExtension(ConfigurationElement.java:286)
	at org.eclipse.core.internal.registry.ConfigurationElementHandle.createExecutableExtension(ConfigurationElementHandle.java:65)
	at org.eclipse.core.internal.filebuffers.ExtensionsRegistry.getExtension(ExtensionsRegistry.java:228)
	at org.eclipse.core.internal.filebuffers.ExtensionsRegistry.doGetDocumentSetupParticipants(ExtensionsRegistry.java:360)
	at org.eclipse.core.internal.filebuffers.ExtensionsRegistry.getDocumentSetupParticipants(ExtensionsRegistry.java:384)
	at org.eclipse.core.internal.filebuffers.ResourceExtensionRegistry.getDocumentSetupParticipants(ResourceExtensionRegistry.java:131)
	at org.eclipse.core.internal.filebuffers.ResourceTextFileBufferManager.createEmptyDocument(ResourceTextFileBufferManager.java:149)
	at org.eclipse.core.internal.filebuffers.ResourceTextFileBuffer.initializeFileBufferContent(ResourceTextFileBuffer.java:246)
	at org.eclipse.core.internal.filebuffers.ResourceFileBuffer.create(ResourceFileBuffer.java:248)
	at org.eclipse.core.internal.filebuffers.TextFileBufferManager.connect(TextFileBufferManager.java:118)
	at org.eclipse.jdt.ls.core.internal.handlers.JsonRpcHelpers.toDocument(JsonRpcHelpers.java:203)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceDiagnosticsHandler.visit(WorkspaceDiagnosticsHandler.java:201)
	at org.eclipse.core.internal.events.ResourceDelta.accept(ResourceDelta.java:78)
	at org.eclipse.core.internal.events.ResourceDelta.accept(ResourceDelta.java:92)
	at org.eclipse.core.internal.events.ResourceDelta.accept(ResourceDelta.java:92)
	at org.eclipse.core.internal.events.ResourceDelta.accept(ResourceDelta.java:61)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceDiagnosticsHandler.resourceChanged(WorkspaceDiagnosticsHandler.java:114)
	at org.eclipse.core.internal.events.NotificationManager$1.run(NotificationManager.java:342)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.core.internal.events.NotificationManager.notify(NotificationManager.java:331)
	at org.eclipse.core.internal.events.NotificationManager.broadcastChanges(NotificationManager.java:185)
	at org.eclipse.core.internal.resources.Workspace.broadcastPostChange(Workspace.java:473)
	at org.eclipse.core.internal.resources.Workspace.endOperation(Workspace.java:1619)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2522)
	at org.eclipse.core.internal.events.NotificationManager$NotifyJob.run(NotificationManager.java:68)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

Generally, the bundle `start()` method should not rely on being called from the UI thread. UI code should run in UI jobs, ideally if the workbench is available.

--- a/org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/TMUIPlugin.java
+++ b/org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/TMUIPlugin.java
@@ -23,6 +23,7 @@ import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Platform;
 import org.eclipse.core.runtime.Status;
 import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jface.resource.ImageRegistry;
 import org.eclipse.jface.text.templates.TemplateContextType;
 import org.eclipse.jface.text.templates.persistence.TemplateStore;
 import org.eclipse.text.templates.ContextTypeRegistry;
@@ -170,8 +171,11 @@ public class TMUIPlugin extends AbstractUIPlugin {
 				}
 			});
 		}
+	}
 
-		TMImages.initalize(getImageRegistry());
+	@Override
+	protected void initializeImageRegistry(final ImageRegistry registry) {
+		TMImages.initalize(registry);
 	}
 
 	@Override

